app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -252,15 +241,24 @@ def save_network_ports_inventory(ems, hashes, target = nil, mode = :refresh)
     hashes.compact!
 
     hashes.each do |h|
-      %i(cloud_tenant device cloud_subnet).each do |relation|
-        h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
+      device = h.fetch_path(:device)
+      if device.kind_of?(Hash)
+        h.delete(:device)
+
+        h[:device_id]   = device[:id]
+        h[:device_type] = device[:type].constantize.base_class.name
       end
 
-      h[:security_groups] = (h.fetch_path(:security_groups) || []).map { |x| x.try(:[], :_object) }.compact.uniq
+      h[:security_group_ids] = (h.delete(:security_groups) || []).map { |x| x.try(:[], :id) }.compact.uniq
```

Comment:
Can can there be nils in here? `{:security_groups => [nil]}`
I know you did not introduce the `try`, but curious why it is there. (not asking for a change)